src/main/java/io/cdap/delta/bigquery/BigQueryEventConsumer.java

```diff
@@ -276,7 +281,13 @@ public synchronized void applyDDL(Sequenced<DDLEvent> sequencedEvent) throws Exc
           primaryKeyStore.put(tableId, primaryKeys);
           context.putState(String.format("bigquery-%s-%s", tableId.getDataset(), tableId.getTable()),
                            Bytes.toBytes(GSON.toJson(new BigQueryTableState(primaryKeys))));
-          TableInfo tableInfo = TableInfo.newBuilder(tableId, tableDefinition).build();
+
+          TableInfo.Builder builder = TableInfo.newBuilder(tableId, tableDefinition);
+          if (cmekKey != null) {
+            builder.setEncryptionConfiguration(
+              EncryptionConfiguration.newBuilder().setKmsKeyName(cmekKey).build());
```

Comment:
can we set this EncryptionConfiguration in the constructor and re-use it?

Reply:
sure, we can do that.